daml-lf/interface/src/main/scala/com/digitalasset/daml/lf/iface/DefDataType.scala

```diff
@@ -173,11 +173,30 @@ final case class Enum(constructors: ImmArraySeq[Ref.Name]) extends DataType[Noth
   def asDataType[RT, PVT]: DataType[RT, PVT] = this
 }
 
-final case class DefTemplate[+Ty](choices: Map[Ref.Name, TemplateChoice[Ty]], key: Option[Ty]) {
+final case class DefTemplate[+Ty](
+    choices: Map[Ref.ChoiceName, TemplateChoice[Ty]],
+    inheritedChoices: Map[Ref.ChoiceName, Ref.TypeConName],
```

Comment:
The "goal" is for this to be empty; I haven't provided any way to represent that phase transition for an `EnvironmentInterface` at the type level in this PR.

Reply:
we should probably call this differently i.e. `unresolvedInheritedChoices`.

Reply:
a75fe7624420dc4c4eef518c29bc192100bbc150